### PR TITLE
Issue 161 add sign in to header

### LINKED ISSUE #161: Add Sign In Form to the Header

### PR DESCRIPTION
**Pull request by chrismborland**
Fix #161 

## 🔗 Test URLs:
  
- Before: https://main--caesars--hlxsites.hlx.page/caesars-palace/
- After: https://issue-161-add-sign-in-to-header--caesars--hlxsites.hlx.page/caesars-palace/

## 📝 Description:
  
- Integrated the sign in form into the header.
- Form is only embedded once and is responsive.

### CODE REVIEW

File: caesars-palace/blocks/form/form.css [hunk 1]
```diff
@@ -33,29 +33,29 @@ main .form select {
...
 
-main .form.sign-in input,
-main .form.sign-in textarea,
-main .form.sign-in select {
+.form.sign-in input,
+.form.sign-in textarea,
+.form.sign-in select {
   padding: 16px 16px 4px;
-  font: inherit;
+  font: 14px;
```
Comment (tmathern) [inline caesars-palace/blocks/form/form.css:44]: Use the CSS variable here if you can

File: caesars-palace/blocks/form/form.css [hunk 2]
```diff
@@ -33,29 +33,29 @@ main .form select {
...
 }
 
-main .form.sign-in input.error,
-main .form.sign-in textarea.error,
-main .form.sign-in select.error { 
+.form.sign-in input.error,
+.form.sign-in textarea.error,
+.form.sign-in select.error { 
   border-color: #d60000;
```
Comment (tmathern) [inline caesars-palace/blocks/form/form.css:50]: Can you try to keep the notation of colors consistent? All rgb?

File: caesars-palace/blocks/form/form.css [hunk 3]
```diff
@@ -81,42 +81,46 @@ main .form.sign-in label {
   font-weight: 400;
 }
 
-main .form span.input-error {
+.form span.input-error {
   display: none;
   font-size: 11px;
   color: #d60000;
```
Comment (tmathern) [inline caesars-palace/blocks/form/form.css:87]: Please use consistent notation for colors accross the file

File: caesars-palace/blocks/form/form.css [hunk 4]
```diff
@@ -81,42 +81,46 @@ main .form.sign-in label {
...
-main .form.sign-in label.active {
+.form.sign-in label.active {
   color: #000;
   transform: scale(0.71) translateY(-50%) translateY(-11px);
 }
 
-main .form label.required::after {
+.form label.required::after {
   content: "*";
   color: red;
```
Comment (tmathern) [inline caesars-palace/blocks/form/form.css:102]: Consistent color notation...

File: caesars-palace/blocks/form/form.css [hunk 5]
```diff
@@ -81,42 +81,46 @@ main .form.sign-in label {
...
 }
 
-main .form .field-wrapper {
+.form .field-wrapper {
   margin-bottom: 20px;
 }
 
-main .form .form-checkbox-wrapper {
+.form.sign-in .field-wrapper {
+  margin-bottom: 16px;
```
Comment (tmathern) [inline caesars-palace/blocks/form/form.css:115]: Use the CSS variable here if you can

File: caesars-palace/blocks/header/header.js [hunk 6]
```diff
@@ -284,6 +351,17 @@ export default async function decorate(block) {
         toggleNavSectionTitles(localNavTitle, newDiv);
       });
       if (globalNavSections) navSections.append(globalNavSections);
+
+      const userAccountMobile = document.createElement('div');
+      userAccountMobile.classList.add('user-account-mobile');
+      const signInMobile = document.createElement('div');
+      signInMobile.classList.add('sign-in');
+      const signInLink = document.createElement('a');
+      signInLink.textContent = 'Sign Up / Sign In';
```
Comment (tmathern) [inline caesars-palace/blocks/header/header.js:360]: Use a constant for the text?

**Review by tmathern — CHANGES_REQUESTED** [review]
I think it would be better to pick one type of color notation in CSS and use it everywhere in the file.

File: caesars-palace/blocks/form/form.css (hunk 1 above)
Comment (chrismborland) [reply caesars-palace/blocks/form/form.css:44]: Done

File: caesars-palace/blocks/form/form.css (hunk 2 above)
Comment (chrismborland) [reply caesars-palace/blocks/form/form.css:50]: Done

File: caesars-palace/blocks/form/form.css [hunk 7]
```diff
@@ -66,7 +66,7 @@ main .form label {
   font-weight: 800;
 }
 
-main .form.sign-in label {
+.form.sign-in label {
   position: absolute;
   margin-left: 16px;
```
[the remark this answers is not in the document]
Comment (chrismborland) [reply caesars-palace/blocks/form/form.css:71]: I'm not sure which variable you'd like me to use here.

File: caesars-palace/blocks/form/form.css [hunk 8]
```diff
@@ -66,7 +66,7 @@ main .form label {
...
   margin-top: 24px;
```
[the remark this answers is not in the document]
Comment (chrismborland) [reply caesars-palace/blocks/form/form.css:72]: I'm not sure which variable you'd like me to use here.

File: caesars-palace/blocks/form/form.css (hunk 5 above)
Comment (chrismborland) [reply caesars-palace/blocks/form/form.css:115]: I don't see an applicable variable for this.

File: caesars-palace/blocks/header/header.css [hunk 9]
```diff
@@ -194,6 +194,23 @@ header nav .nav-sections .global-nav[aria-hidden=true] {
...
+  min-height: 84px;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+}
+
+header nav[aria-expanded=true] .nav-sections .user-account-mobile::before {
+  content: '';
+  margin-right: 10px;
+  right: 16px;
```
[the remark this answers is not in the document]
Comment (chrismborland) [reply caesars-palace/blocks/header/header.css:208]: I don't see an applicable variable for this.

File: caesars-palace/blocks/form/form.css (hunk 4 above)
Comment (chrismborland) [reply caesars-palace/blocks/form/form.css:102]: Done

File: caesars-palace/blocks/form/form.css (hunk 3 above)
Comment (chrismborland) [reply caesars-palace/blocks/form/form.css:87]: Done

File: caesars-palace/blocks/form/form.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (chrismborland) [reply caesars-palace/blocks/form/form.css:72]: Done

File: caesars-palace/blocks/form/form.css (hunk 7 above)
[the remark this answers is not in the document]
Comment (chrismborland) [reply caesars-palace/blocks/form/form.css:71]: Done

File: caesars-palace/blocks/header/header.js [hunk 10]
```diff
@@ -188,6 +202,47 @@ const showMore = (nav, maxItemsDesktop) => {
...
+    toggleMenu(nav, navSections);
+  }
+}
+
+/**
+ * Creates the user menu
+ * @param {Element} block Header block
+ */
+async function createUserMenu(block) {
+  const userMenu = document.createElement('div');
```
[the remark this answers is not in the document]
Comment (chrismborland) [reply caesars-palace/blocks/header/header.js:227]: Hmm... I think it's a bit easier to read if you keep them all together.  How much of a Lighthouse score increase?  Can you send me a link to a reference?

File: caesars-palace/blocks/header/header.js (hunk 6 above)
Comment (chrismborland) [reply caesars-palace/blocks/header/header.js:360]: Done

**Review by amol-anand — APPROVED** [review]

**Pull request merged**